framework/src/base/NonlinearSystem.C

```diff
@@ -1253,6 +1254,12 @@ NonlinearSystem::computeResidualInternal(Moose::KernelType type)
       const std::vector<MooseSharedPointer<ScalarKernel> > & scalars = _scalar_kernels.getActiveObjects();
       for (const auto & scalar_kernel : scalars)
       {
+        // Don't compute the residual if this is an ODETimeKernel and
+        // we've been asked for KT_NONTIME residuals.
+        ODETimeKernel * ode_time = dynamic_cast<ODETimeKernel *>(scalar_kernel.get());
+        if (type == Moose::KT_NONTIME && ode_time)
+          continue;
+
```

Comment:
OK, I can do this optimization, however this PR fixes the immediate serious bugs with TimeIntegrators for SCALAR variables if you think that would be worth getting merged sooner rather than later?